### ISSUE TITLE
Species-specific effects

### ISSUE THREAD

**Issue by Vexatos** [ForestryMC/ForestryMC#1100]
Added to `IAlleleBeeSpeciesCustom`, I guess. An `addEffect` method allowing setting an `IAlleleBeeSpeciesEffect extends IAlleleBeeEffect` to be bound to a species. with such an effect set, the genome will be unable to get an `IAlleleEffect` bred onto it, it must (and will) stay effectNone. The API methods used for retrieving effects will check for such a specific effect first before checking for the genome's effect. It should display differently in the beealyzer to mark that the effect can't be stripped off the bee.

**Comment by mezz** [comment]
I'm not sure I like changes like this, it makes it more difficult to reason about bees. 
They're already complicated, adding more edge cases is not a good idea.

**Comment by Vexatos** [comment]
This change would, however, allow addon developers to not just create bees with special products but bees with special effects; you'd have a different reason to breed a specific species and _stick with them_ as the effect can't be bred to any "better" bee.

**Comment by temp1011** [comment]
Career bees does this, I didn't look into exactly how but it has [this](https://github.com/rwtema/Careerbees/blob/fe5384aed416cc4fe66531ed9674fa368348de56/src/main/java/com/rwtema/careerbees/effects/ISpecialBeeEffect.java) which I assume is called to check if the bee can produce the effect.

**Comment by Vexatos** [comment]
That implementation is exactly the type of hack that I'd like to avoid, considering that transferring the effect is still possible, but won't do anything, without explaining why.

**Comment by temp1011** [comment]
True, however not being able to transfer an effect goes against the basic ideas of bee breeding (and I guess the underlying genetics) that things can be inherited by the offspring.
